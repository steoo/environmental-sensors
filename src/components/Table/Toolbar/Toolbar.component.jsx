import React, { useEffect, useState } from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import { TablePagination, Select, InputLabel, MenuItem } from '@material-ui/core';
import { getFilterBy, getNextPage, getPrevPage, getSortBy } from '../../../store/actions';
import { DEFAULT_LIMIT, DEFAULT_SORT } from '../../../api/constants';

const FlexContainer = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
`;

const SelectContainer = styled(FlexContainer)`
  margin-left: 10px;
`;

const Toolbar = ({ currentPage, count, dispatch }) => {
  const [sortBy, setSortBy] = useState(DEFAULT_SORT);
  const [filterBy, setFilterBy] = useState('');

  const onChangePage = (event, page) => {
    if (page < currentPage - 1) {
      return dispatch(getPrevPage());
    }

    return dispatch(getNextPage());
  };

  const onChangeSort = (event) => {
    setSortBy(event.target.value);
  };

  const onChangeFilter = (event) => {
    setFilterBy(event.target.value);
  };

  useEffect(() => {
    dispatch(getSortBy(sortBy));
  }, [sortBy, dispatch]);

  useEffect(() => {
    dispatch(getFilterBy({ filterName: 'sensor_type', filterValue: filterBy }));
  }, [filterBy, dispatch]);

  const isVisible = currentPage && count;

  return isVisible ? (
    <FlexContainer>
      <FlexContainer>
        <FlexContainer>
          <InputLabel id="sort-by-select">Sort By:</InputLabel>
          <Select labelId="sort-by-select" onChange={onChangeSort} value={sortBy}>
            <MenuItem value="reading_ts">Timestamp</MenuItem>
            <MenuItem value="sensor_type">Sensor Type</MenuItem>
          </Select>
        </FlexContainer>
        <SelectContainer>
          <InputLabel id="filter-by-select">Filter By Sensor Type:</InputLabel>
          <Select labelId="filter-by-select" onChange={onChangeFilter} value={filterBy}>
            <MenuItem value="">None</MenuItem>
            <MenuItem value="O3">O3</MenuItem>
            <MenuItem value="NO2">NO2</MenuItem>
            <MenuItem value="CO">CO</MenuItem>
            <MenuItem value="TEMP">TEMP</MenuItem>
            <MenuItem value="RH">RH</MenuItem>
          </Select>
        </SelectContainer>
      </FlexContainer>
      <TablePagination
        component="div"
        page={currentPage - 1}
        count={count}
        rowsPerPage={DEFAULT_LIMIT}
        rowsPerPageOptions={[DEFAULT_LIMIT]}
        onChangePage={onChangePage}
      />
    </FlexContainer>
  ) : null;
};

Toolbar.propTypes = {
  currentPage: PropTypes.number.isRequired,
  count: PropTypes.number.isRequired,
};

export default Toolbar;
