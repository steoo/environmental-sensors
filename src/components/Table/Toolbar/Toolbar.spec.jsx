import React from 'react';
import { render } from '@testing-library/react';
import Toolbar from './Toolbar.component';

const toolbarData = {
  currentPage: 1,
  count: 800,
  dispatch: jest.fn(),
};

describe('Components', () => {
  describe('Toolbar', () => {
    it('should render, without crashing', () => {
      const { container } = render(<Toolbar {...toolbarData} />);

      expect(container).toMatchSnapshot();
    });
  });
});
