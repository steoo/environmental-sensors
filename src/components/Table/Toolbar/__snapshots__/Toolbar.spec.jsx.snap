// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components Toolbar should render, without crashing 1`] = `
<div>
  <div
    class="sc-bdfBwQ sc-gsTCUz ilXUMA jNCWBR"
  >
    <div
      class="sc-bdfBwQ ilXUMA"
    >
      <div
        class="MuiFormControl-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
          data-shrink="true"
          id="sort-by-select"
        >
          Sort By
        </label>
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        >
          <div
            aria-haspopup="listbox"
            aria-labelledby="sort-by-select"
            class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
            role="button"
            tabindex="0"
          >
            Timestamp
          </div>
          <input
            aria-hidden="true"
            class="MuiSelect-nativeInput"
            tabindex="-1"
            value="reading_ts"
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSelect-icon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </div>
      </div>
      <div
        class="sc-bdfBwQ sc-dlfnbm ilXUMA bEGaqS"
      >
        <div
          class="MuiFormControl-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
            data-shrink="true"
            id="filter-by-select"
          >
            Sensor Type
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
          >
            <div
              aria-haspopup="listbox"
              aria-labelledby="filter-by-select"
              class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
              role="button"
              tabindex="0"
            >
              None
            </div>
            <input
              aria-hidden="true"
              class="MuiSelect-nativeInput"
              tabindex="-1"
              value="none"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSelect-icon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiTablePagination-root"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular MuiTablePagination-toolbar MuiToolbar-gutters"
      >
        <div
          class="MuiTablePagination-spacer"
        />
        <p
          class="MuiTypography-root MuiTablePagination-caption MuiTypography-body2 MuiTypography-colorInherit"
        >
          1-100 of 800
        </p>
        <div
          class="MuiTablePagination-actions"
        >
          <button
            aria-label="Previous page"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit Mui-disabled Mui-disabled"
            disabled=""
            tabindex="-1"
            title="Previous page"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                />
              </svg>
            </span>
          </button>
          <button
            aria-label="Next page"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
            tabindex="0"
            title="Next page"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
